### ISSUE TITLE
Add timeout to external requests

### ISSUE THREAD

**Issue by nuest** [o2r-project/o2r-meta#48]
External requests can be disabled fully (`-xo`) for testing. For production installations we do not want to miss this feature, but must make sure the user gets a reasonable response time. Therefore it would be useful to have a `timeout` option (in seconds, let's not get so picky to use millis) that defines the maximum time that requests to external resources may take.

A timeout is already used, see https://github.com/o2r-project/o2r-meta/blob/dev/extract/metaextract.py#L103, but not exposed to the user of the CLI tool.

A reasonable default could be `5` seconds.

**Comment by 7048730** [comment]
good idea, should be quite easy to add this. 5 secs seems rather low from what I saw on orcid api response time. it sticks around 10-20 secs per request. maybe they do have a premium api for members.
timeout 5s will result empty orcids 80% of the time, I fear.

**Comment by nuest** [comment]
I've just run a stupid test with the same request, 100 times and it took 35 seconds, so less than have a second per request, using Postman. I wanted to see if there is a difference between JSON and ORCID XML - there is none. So no solution there... can you maybe log the response headers to find out more why this is taking so long?

**Comment by 7048730** [comment]
I think because its search `name` and it generates a lot of hits

**Comment by nuest** [comment]
The search uses only `q`, and the problem is not there when accessing the API from other tools:

```
$ ab -n 30 https://pub.orcid.org/v2.0/search?q='Ted Tester'
This is ApacheBench, Version 2.3 <$Revision: 1706008 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking pub.orcid.org (be patient).....done


Server Software:        nginx/1.4.6
Server Hostname:        pub.orcid.org
Server Port:            443
SSL/TLS Protocol:       TLSv1.2,ECDHE-RSA-AES256-GCM-SHA384,2048,256

Document Path:          /v2.0/search?q=Ted Tester
Document Length:        0 bytes

Concurrency Level:      1
Time taken for tests:   15.700 seconds
Complete requests:      30
Failed requests:        0
Non-2xx responses:      30
Total transferred:      6060 bytes
HTML transferred:       0 bytes
Requests per second:    1.91 [#/sec] (mean)
Time per request:       523.328 [ms] (mean)
Time per request:       523.328 [ms] (mean, across all concurrent requests)
Transfer rate:          0.38 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:      379  388  15.7    383     452
Processing:   127  135  12.4    129     184
Waiting:      126  135  12.3    129     183
Total:        506  523  23.3    515     597

Percentage of the requests served within a certain time (ms)
  50%    515
  66%    520
  75%    523
  80%    534
  90%    577
  95%    581
  98%    597
  99%    597
 100%    597 (longest request)
```

**Comment by 7048730** [comment]
When I limited the requests timeout to 3 seconds in my tests, the each request was answered in 1 second. This could be due to my setup or their own updates. Anyway I think we can live with that now.  I would suggest to leave the timeout hardcoded since a config for this detail seem to be a little bit over the top: the orcid is a suggestion from the extractor anyway.

**Comment by 7048730** [comment]
included with 1f627ae894602218c99a2ec8859c5d0430e17b98
